test/format-test.cc

```diff
@@ -2507,3 +2509,67 @@ TEST(format_test, writer) {
   fmt::writer(s).print("foo");
   EXPECT_EQ(s.str(), "foo");
 }
+
+#if FMT_USE_BITINT
+#  pragma clang diagnostic push
+#  pragma clang diagnostic ignored "-Wbit-int-extension"
+
+template <size_t N, bool is_signed>
+using bitint_helper =
+    fmt::conditional_t<is_signed, _BitInt(N), unsigned _BitInt(N)>;
+template <size_t N> using signed_bitint = bitint_helper<N, true>;
+template <size_t N> using unsigned_bitint = bitint_helper<N, false>;
+
+#  if FMT_USE_BITINT
+TEST(format_test, bitint) {
+  EXPECT_EQ(fmt::format("{}", unsigned_bitint<3>{7}), "7");
+  EXPECT_EQ(fmt::format("{}", signed_bitint<7>{}), "0");
+
+  EXPECT_EQ(fmt::format("{}", unsigned_bitint<15>{31000}), "31000");
+  EXPECT_EQ(fmt::format("{}", signed_bitint<16>{INT16_MIN}), "-32768");
+  EXPECT_EQ(fmt::format("{}", signed_bitint<16>{INT16_MAX}), "32767");
+
+  EXPECT_EQ(fmt::format("{}", unsigned_bitint<32>{4294967295}), "4294967295");
+
+  EXPECT_EQ(fmt::format("{}", unsigned_bitint<47>{140737488355327ULL}),
+            "140737488355327");
+  EXPECT_EQ(fmt::format("{}", signed_bitint<47>{-40737488355327LL}),
+            "-40737488355327");
+
+  // Check lvalues and const
+  auto a = signed_bitint<8>{0};
+  auto b = unsigned_bitint<32>{4294967295};
+  const auto c = signed_bitint<7>{0};
+  const auto d = unsigned_bitint<32>{4294967295};
+  EXPECT_EQ(fmt::format("{}", a), "0");
+  EXPECT_EQ(fmt::format("{}", b), "4294967295");
+  EXPECT_EQ(fmt::format("{}", c), "0");
+  EXPECT_EQ(fmt::format("{}", d), "4294967295");
+
+  static_assert(fmt::is_formattable<signed_bitint<64>, char>{}, "");
+  static_assert(fmt::is_formattable<unsigned_bitint<64>, char>{}, "");
+
+#    if FMT_USE_INT128
+  static_assert(fmt::is_formattable<signed_bitint<128>, char>{}, "");
+  static_assert(fmt::is_formattable<unsigned_bitint<128>, char>{}, "");
+
+  EXPECT_EQ(fmt::format("{}", signed_bitint<128>(0)), "0");
+  EXPECT_EQ(fmt::format("{}", unsigned_bitint<128>(0)), "0");
+  EXPECT_EQ("9223372036854775808",
+            fmt::format("{}", signed_bitint<65>(INT64_MAX) + 1));
```

Comment:
Let's use `numeric_limits` instead of C macros.

Reply:
> Let's use `numeric_limits` instead of C macros.

Sure. I assumed you'd be fine with these since I just copied them from higher up in this file

https://github.com/fmtlib/fmt/blob/e3676ca309fd6c128250f65cd4d32bd9f22e2ef9/test/format-test.cc#L1254-L1262

Reply:
If we already use macros elsewhere then it's fine to keep them here.